### ISSUE TITLE
@text-color-inverse should be dark in the dark mode theme

### ISSUE THREAD

**Issue by KrishnaPG** [ant-design/ant-design#29424]
- [x ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently the `@text-color-inverse` is mapped to be white in the `dark mode` style file (see below):

https://github.com/ant-design/ant-design/blob/1c8a88b2d7631129d0c6707e85fd451dd50b9a3e/components/style/themes/dark.less#L169 

In the `dark` mode, the `inverse` should be `black` and not  `white`.

-------------
In other words, currently `@text-color-inverse` is mapped to be `white` in **both** _default_ and _dark_ mode styles, which is incorrect. Below is the _default_ style where it is also mapped as `white`:
https://github.com/ant-design/ant-design/blob/1c8a88b2d7631129d0c6707e85fd451dd50b9a3e/components/style/themes/default.less#L55 

### What does the proposed API look like?
Please correct the _dark_ mode style file to have  `@text-color-inverse` map to **black** correctly

**Comment by zombieJ** [comment]
Hi,
Could you help to provide the bug of current style color?

**Comment by KrishnaPG** [comment]
When building text highlight interfaces (such as [`annotators`](https://github.com/hypothesis/h)), it is usual to highlight portions of text with some background color and then use `inverse color` to make the text visible. 

A **BUG** is happening because currently `@text-color-inverse` is mapped to be `white` (same as `@text-color`)  **incorrectly** in the `style/themes/dark.less` which produces incorrect results as shown below:

![image](https://user-images.githubusercontent.com/4610831/109404791-8430b180-798f-11eb-91ea-aa3ed6e257cf.png)

Below is how it should be if `@text-color-inverse` is correctly defined to be **inverse** of `@text-color`:

![image](https://user-images.githubusercontent.com/4610831/109404761-2308de00-798f-11eb-8e07-0a468dd2ea5d.png)


**Question / Suggestion**:
Instead of hard-coding the `@text-color-inverse` as `@white`  or `@black`, is it possible to declare it as the inverse, something like
````
 @text-color-inverse: inverse(@text-color); 
````
Not sure if such function `inverse` exists, but that should be the right way of defining the color correctly semantically.

**Comment by jaylonez** [comment]
`@text-color-inverse: @white - @text-color;` would work to map it to the opposite of @text-color.

Right now it seems like the same @text-color-inverse variable is being used for 2 slightly different use-cases: highlight text color and text within components with background color.

What about creating a new variable:
```
{
  @text-color-inverse: @black;
  @text-color-highlighted: @white - @text-color
}
```

**Comment by KrishnaPG** [comment]
Thank you @jaylonez 

> @text-color-inverse: @white - @text-color; would work to map it to the opposite of @text-color.

Would this equation `@white - @text-color` work correctly for both _dark_ style and _light_ styles ? It seems it does, but want to confirm.

In the light style,  `@text-color-inverse` is correctly defined as `white`, so it seems the intention is to have it to be opposite of `@text-color`. Hence semantically it would not seem appropriate to leave it as `@text-color-inverse: @white` in the dark style when creating a new variable.

Did a quick search in the code where the inverse is being used, and these are some of the additional buggy scenarios (not sure if they are related to text-inverse directly, though, but needs fixing in general):

![image](https://user-images.githubusercontent.com/4610831/109441659-d63f0900-7a5b-11eb-9b47-f93f722408f9.png)

![image](https://user-images.githubusercontent.com/4610831/109441681-ebb43300-7a5b-11eb-9a85-323361854bda.png)

![image](https://user-images.githubusercontent.com/4610831/109441750-274efd00-7a5c-11eb-8d5e-de7b07ef732e.png)

![image](https://user-images.githubusercontent.com/4610831/109441788-40f04480-7a5c-11eb-8564-1ed10a141ffd.png)

**Comment by zombieJ** [comment]
`@text-color-inverse` is used for the component with background color. It's not related with theme:
<img width="103" alt="截屏2021-02-27 下午11 13 00" src="https://user-images.githubusercontent.com/5378891/109391304-575da980-7951-11eb-9207-b0ee9857dc12.png">

In dark theme, primary color is same as the light one:
<img width="580" alt="截屏2021-03-01 下午2 56 11" src="https://user-images.githubusercontent.com/5378891/109462326-453d5180-7a9e-11eb-9d0d-4f3347e44072.png">

It's by design to make `@text-color-inverse` as white since we do not modify selection color. So when you customize related less variable. You can change `@text-color-inverse` on your side : )

**Comment by jaylonez** [comment]
@KrishnaPG What less variable settings did you use to get these screenshots?

**Comment by KrishnaPG** [comment]
Thank you @zombieJ 

In that case perhaps the name `@text-color-inverse` is confusing, since it is suggesting that it is **opposite** of `@text-color`.

Is there any built-in variable that we can use to be always opposite of text color? Perhaps it could be defined to be `@white - @text-color ` as @jaylonez  suggested.

@jaylonez These later screenshots were from https://antdtheme.com/dark default settings.

**Comment by zombieJ** [comment]
Yes. `@text-color-inverse` comes earlier than dark theme years ago. I agree it's confused but may not modify its name to avoid breaking change.

**Comment by KrishnaPG** [comment]
Understood. Thank you @zombieJ
